Sources/SentrySwiftUI/SentryTracedView.swift

```diff
@@ -38,62 +40,122 @@ import SwiftUI
+    /// - Parameter content: The content that you want to track the performance
+    public init(_ viewName: String? = nil, waitForFullDisplay: Bool? = nil, @ViewBuilder content: @escaping () -> Content) {
         self.content = content
         self.name = viewName ?? SentryTracedView.extractName(content: Content.self)
         self.nameSource = viewName == nil ? .component : .custom
+        self.waitforFullDisplay = waitForFullDisplay ?? SentrySDK.options?.enableTimeToFullDisplayTracing ?? false
     }
-
+#else
+    /// Creates a view that measures the performance of its `content`.
+    ///
+    /// - Parameter viewName: The name that will be used for the span, if nil we try to get the name of the content class.
+    /// - Parameter content: The content that you want to track the performance
+    public init(_ viewName: String? = nil, @ViewBuilder content: @escaping () -> Content) {
+        self.content = content
+        self.name = viewName ?? SentryTracedView.extractName(content: Content.self)
+        self.nameSource = viewName == nil ? .component : .custom
+    }
+#endif
+    
     private static func extractName(content: Any) -> String {
         var result = String(describing: content)
-
+        
         if let index = result.firstIndex(of: "<") {
             result = String(result[result.startIndex ..< index])
         }
-
+        
         return result
     }
-
+    
     public var body: some View {
-        if viewAppeared {
-            return self.content().onAppear()
-        }
-
-        var transactionCreated = false
-        if SentryPerformanceTracker.shared.activeSpanId() == nil {
-            transactionCreated = true
-            let transactionId = SentryPerformanceTracker.shared.startSpan(withName: self.name, nameSource: nameSource, operation: "ui.load", origin: self.traceOrigin)
-            SentryPerformanceTracker.shared.pushActiveSpan(transactionId)
-
-            //According to Apple's documentation, the call to `body` needs to be fast
-            //and can be made many times in one frame. Therefore they don't use async code to process the view.
-            //Scheduling to finish the transaction at the end of the main loop seems the least hack solution right now.
-            //'onAppear' is not a suitable place to do this because it may happen before other view `body` property get called.
-            DispatchQueue.main.async {
-                SentryPerformanceTracker.shared.popActiveSpan()
-                SentryPerformanceTracker.shared.finishSpan(transactionId)
-            }
+        var trace: SentryTracer?
+        var spanId: SpanId?
+        
+        if !viewAppeared {
+            trace = ensureTransactionExists()
+            spanId = createAndPushBodySpan(transactionCreated: trace != nil)
+#if canImport(SwiftUI) && canImport(UIKit) && os(iOS) || os(tvOS)
+            if let trace = trace { startTTDTraker(for: trace) }
+#endif
```

Comment:
`m`: I would prefer keeping the logic for creating the transaction or span and keeping track of the TTDTracker in the `SentryUIViewControllerPerformanceTracker`. Now, we have two different places that can easily get out of sync.

Reply:
This specific piece of logic is particular to the SwiftUI lifecycle. I don’t think it makes sense to move it to SentryUIViewControllerPerformanceTracker, but I did move other parts related to TTID to SentryUIViewControllerPerformanceTracker to reduce the responsibilities of this class.